liota/dccs/iotcc.py

```diff
@@ -332,16 +331,33 @@ def _create_iotcc_json(self):
         return iotcc_path
 
     def store_reg_entity_details(self, entity_type, entity_name, reg_entity_id, entity_local_uuid):
-        msg = ''
         if self._iotcc_json == '':
             log.warn('iotcc.json file missing')
             return
         try:
-            with open(self._iotcc_json, 'r') as f:
-                msg = json.load(f)
-            f.close()
+            f = open(self._iotcc_json, 'r')
         except IOError, err:
-            log.error('Could not open {0} file '.format(self._iotcc_json) + str(err))
+            log.exception('Could not open {0} file '.format(self._iotcc_json) + str(err))
+            return
+
+        def load_json_record(f):
```

Comment:
An alternative mechanism to read iotcc_json file should be explored if the issue doesn`t get fixed with the retry mechanism.

Reply:
from my test, resolved. we only call this func when register edge system. but will see QE result.